src/main/java/org/springframework/integration/kafka/dsl/KafkaMessageDrivenChannelAdapterSpec.java

```diff
@@ -141,6 +142,17 @@ public S recoveryCallback(RecoveryCallback<? extends Object> recoveryCallback) {
 		return _this();
 	}
 
+	/**
+	 * When using a type-aware message converter (such as {@code StringJsonMessageConverter},
+	 * set the payload type the converter should create. Defaults to {@link Object}.
+	 * @param payloadType the type.
+	 * @return the spec
+	 */
+	public S payloadType(Class<V> payloadType) {
```

Comment:
No, this is not about `V`. This is a fallback type to which that `V` is going to be converted.
Well, at runtime it really doesn't matter because this generic type is going to be erased anyway.
But for the proper code logic we need to be explicit what we are going to do here.

Please, add also `@since` to this new public API.

Also would be great to have some test-case to cover such a functionality.

Reply:
I took a guess on the version number. Is 3.2.0 correct?

Reply:
Correct. Thank you for reminding about that! 😉 
It is really hard to keep in mind all the versions we support in all our projects.

Well, we always can find the current version in the `gradle.properties`.

Reply:
I've pushed a change with the version number.